src/cpp/fastrtps_deprecated/types/TypeIdentifier.cpp

```diff
@@ -1304,6 +1310,119 @@ bool TypeIdentifier::operator==(const TypeIdentifier &other) const
+            }
+            else if (_d() == TI_STRING16_SMALL && x._d() == TI_STRING16_SMALL)
+            {
+                return this->string_sdefn().consistent(x.string_sdefn(), consistency);
+            }
+            else if (_d() == TI_STRING16_SMALL && x._d() == TI_STRING16_LARGE)
+            {
+                return this->string_sdefn().consistent(x.string_ldefn(), consistency);
+            }
+            else if (_d() == TI_STRING8_LARGE && x._d() == TI_STRING8_SMALL)
+            {
+                return this->string_ldefn().consistent(x.string_sdefn(), consistency);
+            }
+            else if (_d() == TI_STRING8_LARGE && x._d() == TI_STRING8_LARGE)
+            {
+                return this->string_ldefn().consistent(x.string_ldefn(), consistency);
+            }
+            else if (_d() == TI_STRING16_LARGE && x._d() == TI_STRING16_SMALL)
+            {
+                return this->string_ldefn().consistent(x.string_sdefn(), consistency);
+            }
+            else if (_d() == TI_STRING16_LARGE && x._d() == TI_STRING16_LARGE)
+            {
+                return this->string_ldefn().consistent(x.string_ldefn(), consistency);
+            }
+            else
+            {
+                return false;
+            }
+
+        case TI_PLAIN_SEQUENCE_SMALL:
+            return this->seq_sdefn().consistent(x.seq_sdefn(), consistency);
+
+        case TI_PLAIN_SEQUENCE_LARGE:
+            return this->seq_ldefn().consistent(x.seq_ldefn(), consistency);
+
+        case TI_PLAIN_ARRAY_SMALL:
+            return this->array_sdefn().consistent(x.array_sdefn(), consistency);
+
+        case TI_PLAIN_ARRAY_LARGE:
+            return this->array_ldefn().consistent(x.array_ldefn(), consistency);
+
+        case TI_PLAIN_MAP_SMALL:
+            return this->map_sdefn().consistent(x.map_sdefn(), consistency);
+
+        case TI_PLAIN_MAP_LARGE:
+            return this->map_ldefn().consistent(x.map_ldefn(), consistency);
+
+        case EK_MINIMAL:
+        case EK_COMPLETE:
+        {
+            if (this->_d() != x._d())
+            {
+                return false;
+            }
+
+            const TypeObject* localObj = TypeObjectFactory::get_instance()->get_type_object(this);
+            const TypeObject* remoteObj = TypeObjectFactory::get_instance()->get_type_object(&x);
+            if (localObj == nullptr)
+            {
+                logWarning(XTYPES, "Local TypeIdentifier doesn't have a related TypeObject");
+                return false;
+            }
+            if (remoteObj == nullptr)
+            {
+                logWarning(XTYPES, "Remote TypeIdentifier doesn't have a related TypeObject");
+                return false;
+            }
+            return localObj->consistent(*remoteObj, consistency);
+        }
+        default:
+            break;
+    }
+    return false;
+}
+
 } // namespace types
 } // namespace fastrtps
 } // namespace eprosima
```

Comment:
Just realized that this file and some others above and below have been moved to `fastrtps_deprecated`, is that what we want? I thought we would have in that folder the current DDS API that we will eventually remove, but will we also remove classes in `types/`?

Reply:
If I'm not wrong, we decided to apply the following changes to `src/cpp`:
* Move `fastrtps/rtps` to `rtps`
* Move `fastrtps` to `fastrtps_deprecated`.
* Add `fastdds` for the new API implementation.